torch/fx/experimental/symbolic_shapes.py

```diff
@@ -3967,12 +3967,20 @@ def _compute_symbolic_stride(
         stride: List[Optional[sympy.Expr]] = [None] * len(size)
         candidates: Dict[Union[int, SymInt], sympy.Expr] = {}
 
-        # iterate over unbound strides in sorted order
-        val_list = [(val, i) for i, val in enumerate(ex_stride)]
+        # iterate over unbound strides in val ascending order with
+        # index descending as a tie breaker since for cases like
+        # [(1, 1), (1, 0)], we want to fill in the right most
+        # stride first.
+        val_list = [(val, -i) for i, val in enumerate(ex_stride)]
         val_list.sort(key=_nested_int_aware_sort)
 
-        for val, i in val_list:
-            if val in (0, 1):
+        for val, neg_i in val_list:
+            i = -neg_i
+            contiguous_stride = (
+                i != len(ex_stride) - 1
+                and ex_stride[i] == ex_size[i + 1] * ex_stride[i + 1]
+            )
```

Comment:
This looks much better. One question I have, though, is there doesn't seem to be unbacked specific here, are you changing the behavior for 1 strides in the non-unbacked case? But maybe this doesn't matter as the size would have already been 1 so you don't pick up variables.